metadata-ingestion/src/datahub/ingestion/source/hex/datahub.py

```diff
@@ -0,0 +1,385 @@
+                        dataset_subjects.append(DatasetUrn.from_string(subject.entity))
+                    elif subject.entity and subject.entity.startswith(
+                        "urn:li:schemaField:"
+                    ):
+                        schema_field_subjects.append(
+                            SchemaFieldUrn.from_string(subject.entity)
+                        )
+
+                if not dataset_subjects and not schema_field_subjects:
+                    self.report.filtered_out_queries_no_subjects += 1
+                    continue
+
+                # Create response
+                response = QueryResponse(
+                    urn=query_urn,
+                    hex_project_id=hex_project_id,
+                    dataset_subjects=dataset_subjects,
+                    schema_field_subjects=schema_field_subjects,
+                )
+                logger.debug(
+                    f"Succesfully extracted {len(dataset_subjects)} dataset subjects and {len(schema_field_subjects)} schema field subjects for query {query_urn}: {dataset_subjects} {schema_field_subjects}"
+                )
+                self.report.total_queries += 1
+                self.report.total_dataset_subjects += len(dataset_subjects)
+                self.report.total_schema_field_subjects += len(schema_field_subjects)
+
+                logger.debug(
+                    f"Processed query {query_urn} with Hex project ID {hex_project_id}"
+                )
+                yield response
+
+    def _fetch_query_entities(self, query_urns: List[QueryUrn]) -> EntitiesResponse:
+        entities_by_urn = EntitiesResponse()
+
+        for i in range(0, len(query_urns), self.page_size):
+            batch = query_urns[i : i + self.page_size]
+
+            logger.debug(f"Fetching query entities for {len(batch)} queries: {batch}")
+            response_json = self.datahub_client._graph.get_entities_v2(
+                entity_name="query",
+                urns=[urn.urn() for urn in batch],
+                aspects=[
+                    QueryPropertiesClass.ASPECT_NAME,
+                    QuerySubjectsClass.ASPECT_NAME,
+                ],
+                with_system_metadata=False,
+            )
+            self.report.num_calls_fetch_query_entities += 1
+            logger.debug(f"Get entities v2 response: {response_json}")
+            batch_entities_by_urn = EntitiesResponse.parse_obj(response_json)
+            entities_by_urn.update(batch_entities_by_urn)
+
+        return entities_by_urn
+
+    def _fetch_query_urns_filter_hex_and_last_modified(self) -> List[QueryUrn]:
+        """
+        This could be implemented as follows, but search sdk still requires some fixes.
+        query_urns = self.datahub_client.search.get_urns(
+            filter=F.and_(
+                F.entity_type(QueryUrn.ENTITY_TYPE),
+                F.custom_filter("origin", "EQUAL", [HEX_PLATFORM_URN.urn()]),
+                F.custom_filter(
```

Comment:
with my fixes in https://github.com/datahub-project/datahub/pull/13026, we should be able to use this again

Reply:
addressed in commit a736a76ebbe9f66e254d9ac93bf1d1073a84e66c